--- configure.orig	2007-03-27 02:42:13.000000000 -0400
+++ configure	2007-03-27 04:37:03.000000000 -0400
@@ -18,8 +18,8 @@
 EXE_EXT=
 
 CC=gcc
-CFLAGS_DEMO="-Iinclude -O2 -Wall"
-CFLAGS="-Iinclude -O2 -Wall" 
+CFLAGS_DEMO="$CFLAGS -Iinclude -Wall"
+CFLAGS="$CFLAGS -Iinclude -Wall" 
 CFLAGS_ADD='-o $@'
 CFLAGS_EXE='-o '
 CFLAGS_USER=
@@ -270,22 +270,22 @@
 
 	# commands to install #
 	echo "INCDIR=${PREFIX}/include"
-	echo "LIBDIR=${PREFIX}/lib"
-	INSTALL_CMD0='if [ ! -d $(PREFIX) ]; then mkdir -p $(PREFIX); fi'
-	INSTALL_CMD1='if [ ! -d $(PREFIX)/include ]; then mkdir -p $(PREFIX)/include; fi'
-	INSTALL_CMD2='if [ ! -d $(PREFIX)/lib ]; then mkdir -p $(PREFIX)/lib; fi'
+	echo "LIBDIR=${PREFIX}/${LIBDIR}"
+	INSTALL_CMD0='if [ ! -d ${DESTDIR}$(PREFIX) ]; then mkdir -p ${DESTDIR}$(PREFIX); fi'
+	INSTALL_CMD1='if [ ! -d ${DESTDIR}$(PREFIX)/include ]; then mkdir -p ${DESTDIR}$(PREFIX)/include; fi'
+	INSTALL_CMD2='if [ ! -d ${DESTDIR}$(PREFIX)/${LIBDIR} ]; then mkdir -p ${DESTDIR}$(PREFIX)/${LIBDIR}; fi'
 	if test $SHARED = yes; then
-		INSTALL_CMD3='cp include/hpdf.h include/hpdf_consts.h include/hpdf_types.h $(PREFIX)/include/; chmod 644 $(PREFIX)/include/hpdf*.h'
-		INSTALL_CMD4='cp -p $(SONAME)$(SOVER1)$(SOVER2) $(PREFIX)/lib/'
+		INSTALL_CMD3='cp include/hpdf*.h ${DESTDIR}$(PREFIX)/include/; chmod 644 ${DESTDIR}$(PREFIX)/include/hpdf*.h'
+		INSTALL_CMD4='cp -p $(SONAME)$(SOVER1)$(SOVER2) ${DESTDIR}$(PREFIX)/${LIBDIR}'
 		if test $SHARED_EXT != '.dll'; then
-			INSTALL_CMD5='ln -sf $(SONAME)$(SOVER1)$(SOVER2) $(PREFIX)/lib/$(SONAME)$(SOVER1)'
-			INSTALL_CMD6='ln -sf $(SONAME)$(SOVER1) $(PREFIX)/lib/$(SONAME)'
+			INSTALL_CMD5='ln -sf $(SONAME)$(SOVER1)$(SOVER2) ${DESTDIR}$(PREFIX)/${LIBDIR}/$(SONAME)$(SOVER1)'
+			INSTALL_CMD6='ln -sf $(SONAME)$(SOVER1) ${DESTDIR}$(PREFIX)/${LIBDIR}/$(SONAME)'
 		else
-			INSTALL_CMD5='cp -p $(LIBNAME) $(PREFIX)/lib/'
+			INSTALL_CMD5='cp -p $(LIBNAME) ${DESTDIR}$(PREFIX)/${LIBDIR}/'
 		fi
 	else
-		INSTALL_CMD3='cp include/hpdf*.h $(PREFIX)/include/; chmod 644 $(PREFIX)/include/hpdf*.h'
-		INSTALL_CMD4='cp -p $(LIBNAME) $(PREFIX)/lib/'
+		INSTALL_CMD3='cp include/hpdf*.h ${DESTDIR}$(PREFIX)/include/; chmod 644 ${DESTDIR}$(PREFIX)/include/hpdf*.h'
+		INSTALL_CMD4='cp -p $(LIBNAME) ${DESTDIR}$(PREFIX)/${LIBDIR}/'
 	fi	
 fi
 
